build/VSUtilities.psm1

```diff
@@ -0,0 +1,119 @@
+function Start-VsDevShell {
+    $chosenVisualStudioVersion = ChooseVisualStudioInstallation @(Get-VSInstalls)
+    if ($chosenVisualStudioVersion -eq $null) {
+            return
+    }
+
+    $installDir = $chosenVisualStudioVersion.InstallationPath
+    $devShellModule = Join-Path $installDir 'Common7\Tools\Microsoft.VisualStudio.DevShell.dll'
+    Import-Module $devShellModule
+    Enter-VsDevShell -DevCmdArguments "-no_logo" -StartInPath "$pwd" $chosenVisualStudioVersion.InstanceId
+}
+
+function Get-VsCommonToolsPath {
+    if([string]::IsNullOrEmpty($env:VSINSTALLDIR)) {
+        $chosenVisualStudioVersion = ChooseVisualStudioInstallation @(Get-VSInstalls)
+        $installDir = $chosenVisualStudioVersion.InstallationPath
+    }
+    else {
+        $installDir = $env:VSINSTALLDIR
+    }
+
+    if([string]::IsNullOrEmpty($installDir)) {
+        return
+    }
+
+    $vsComnToolsPath = Join-Path $installDir 'Common7\Tools\'
+    return $vsComnToolsPath
+}
+
+function Get-VSWhere {
+    $ProgramFile = ${env:ProgramFiles(x86)}
+
+    if ($null -eq $ProgramFile) {
+        $ProgramFile = (Get-ChildItem env:ProgramFiles).Value
+    }
+
+    $vsWhereVersion = "2.8.4"
+    $vsWhereDir = Join-Path $ProgramFile 'Microsoft Visual Studio\Installer'
+    $vsWhereExe = "$vsWhereDir\vswhere.exe"
+    
+    if (!(Test-Path $vsWhereExe)) {
+        if (!(Test-Path $vsWhereDir)) {
+            Write-Host "Creating folder $vsWhereDir"
+            New-Item -ItemType Directory -Path $vsWhereDir
+        }
+
+        Write-Host "Downloading vswhere"
+        [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+        Invoke-WebRequest "http://github.com/Microsoft/vswhere/releases/download/$vsWhereVersion/vswhere.exe" -OutFile $vswhereExe
+    }
+
+    return $vsWhereExe
+}
+
+function Get-VSInstalls {
+
+    $vsWhereExe = Get-VSWhere
+
+    $temp = & "$vsWhereExe" -nologo -all -prerelease -format text | select-string "instanceId|installationPath|installationName|catalog_productSemanticVersion"
```

Comment:
You're probably already aware, but vswhere now supports `-format` to get json or xml, in case that's easier